pkg/operator/ceph/object/rgw.go

```diff
@@ -211,13 +205,20 @@ func (c *clusterConfig) startRGWPods() error {
 			depIDToRemove := currentRgwInstances - 1
 			depNameToRemove := fmt.Sprintf("%s-%s-%s", AppName, c.store.Name, k8sutil.IndexToName(depIDToRemove))
 			if err := k8sutil.DeleteDeployment(c.context.Clientset, c.store.Namespace, depNameToRemove); err != nil {
-				logger.Warning("error during deletion of deployment %q resource. %v", depNameToRemove, err)
+				logger.Warningf("error during deletion of deployment %q resource. %v", depNameToRemove, err)
 			}
 			currentRgwInstances = currentRgwInstances - 1
 			i++
 
+			// Delete the Secret key
+			secretToRemove := fmt.Sprintf("%s-%s-%s-keyring", AppName, c.store.Name, k8sutil.IndexToName(depIDToRemove))
+			err = c.context.Clientset.CoreV1().Secrets(c.store.Namespace).Delete(secretToRemove, &metav1.DeleteOptions{})
```

Comment:
Is the deletion of this secret going to trigger another reconcile?

Reply:
Yes... it's difficult to match it, the object diff for the change is `{"metadata":{"deletionTimestamp":"2020-03-10T14:54:38Z","finalizers":["foregroundDeletion"],"generation":2}}`.

Reply:
I've had something to catch that message before but felt it was too weak and could cause unwanted issues so I removed it...